src/test/java/UseCase/CrearPersonaUseCaseTest.java

```diff
@@ -0,0 +1,117 @@
+
+@ExtendWith(MockitoExtension.class)
+class CrearPersonaUseCaseTest {
+
+    @Mock
+    private GuardarPersona guardarPersona;
+
+
+
+    @Test
+    void CrearPersona_Success() {
+        Persona miPersona = Persona.factory("Juan",
+                "Perez",
+                LocalDate.of(2000, 1, 1),
+                "43611353",
+                180.0f,
+                80.5f);
+        CrearPersonaUseCase personaUseCase = new CrearPersonaUseCase(guardarPersona);
+        when(guardarPersona.validarPersona(miPersona.getDni())).thenReturn(false); // No existe
+        when(guardarPersona.guardarPersona(miPersona)).thenReturn(true); // Guardado exitoso
+
+
+        boolean resultado = personaUseCase.crearPersona(miPersona.getNombre(),
+                miPersona.getApellido(),
+                miPersona.getFechaNacimiento(),
+                miPersona.getDni(),
+                miPersona.getAltura(),
+                miPersona.getPeso());
+        Assertions.assertTrue(resultado);
+
+    }
+
+    @Test
+    void CrearPersona_Duplicada() {
+        Persona miPersona = Persona.factory("Juan",
+                "Perez",
+                LocalDate.of(2000, 1, 1),
+                "43611353",
+                180.0f,
+                80.5f);
+        CrearPersonaUseCase personaUseCase = new CrearPersonaUseCase(guardarPersona);
+        when(guardarPersona.validarPersona("43611353")).thenReturn(true); // Ya existe
+
+        PersonaExisteException exception = Assertions.assertThrows(PersonaExisteException.class, () -> {
+            personaUseCase.crearPersona(miPersona.getNombre(),
+                    miPersona.getApellido(),
+                    miPersona.getFechaNacimiento(),
+                    miPersona.getDni(),
+                    miPersona.getAltura(),
+                    miPersona.getPeso());
+        });
+
+        Assertions.assertEquals("La persona ya existe", exception.getMessage());
+
+        verify(guardarPersona).validarPersona("43611353");
+        verify(guardarPersona, never()).guardarPersona(any(Persona.class));
+    }
+
+    @Test
+    void CrearPersona_ErrorGuardar() {
+        Persona miPersona = Persona.factory("Juan",
+                "Perez",
+                LocalDate.of(2000, 1, 1),
+                "43611353",
+                180.0f,
+                80.5f);
+        CrearPersonaUseCase persona = new CrearPersonaUseCase(guardarPersona);
+        when(guardarPersona.validarPersona("43611353")).thenReturn(false); // No existe
+        when(guardarPersona.guardarPersona(any(Persona.class))).thenReturn(false); // Error al guardar
+
+        PersonaExisteException exception = Assertions.assertThrows(PersonaExisteException.class, () -> {
+            persona.crearPersona(miPersona.getNombre(),
+                    miPersona.getApellido(),
+                    miPersona.getFechaNacimiento(),
+                    miPersona.getDni(),
+                    miPersona.getAltura(),
+                    miPersona.getPeso());
```

Comment:
Solo para limpiar código y evitar procesos adicionales: No es necesario instanciar la persona y usar los getters para llamar al useCase. Es conveniente enviar los valores que pasaría quien llame al usecase

```suggestion
            persona.crearPersona("Juan"
                    "Perez",
                   LocalDate.of(2000, 1, 1),
                   "43611353",
                   180.0f,
                   80.5f);
```

Aplica a todos los test